modules/ROOT/pages/dw-operators.adoc

```diff
@@ -363,3 +363,359 @@ xref:dataweave-variables.adoc#example_local_variable[Examples: Local DataWeave V
 | `else if` | An `else` operator chains expressions together within an if-else construct by incorporating `else if`. See xref:dataweave-flow-control.adoc#control_flow_else_if[else if] for an example.
 |===
+
+== Update Operator
+
+DataWeave 2.0 supports the update operator that enables you to update a given field on a data structure with a new value. This use case is very common and was not simple to do in DataWeave.
+
+_Introduced in DataWeave 2.3.0. Supported by Mule 4.3 and later._ +
+
+Example: Using Update Operator, let's say that you have an object:
+[source,json,linenums]
+----
+{
+    "name": "Ken",
+    "lastName":"Shokida",
+    "age": 30
+}
+----
+
+And you want to update the age by increasing it with one, without the need of recreating all the object again. Using the current DataWeave syntax is:
+
+[source,weave,linenums]
+----
+payload mapObject ((value,key) ->
+    if(key as String == "age")
+       {(key): value as Number + 1}
+    else
+       {(key): value} )
+----
+
+Using the update operator will be:
+
+[source,weave,linenums]
+----
+payload update {
+    case age at .age -> age + 1
+}
+----
+
+As shown in the previous example with the update operator there is no need for casting or going through all the key value pairs. The update syntax is as follows:
+
+[source,weave,linenums]
+----
+<value_to_update> update {
+    case <variable_name> at <update_expression>[!]? [if(<conditional_expression>)]? -> <new value>
+    ...
+}
+----
+
+* `value_to_update` represents the original value that is going to be updated.
+* `update_expression` is the selector expression that is going to match a given value of the `value_to_update`
+* `variable_name` is the name of the variable that is going to bind to the matched `update_expression` existing value.
+* `new_value` is the expression with the new value.
+* `[if(<conditional_expression>)]?` If `conditional_expression` is true then the value is update. This section is optional and only required for Optional update.
+* `[!]` marks then selector as an upsert, which means that if the selector is not matched then it is going to create it.
+* `update_expression` supports multiple `case` expressions.
+
+[NOTE]
+The update operator doesn't mutate the existing value but rather create a new one with the updated expressions.
+
+=== Selector Expressions
+
+Use selector expressions with the update operator.
+
+==== Value Selector
+
+This matching selector allows to match any field value of a given object. +
+
+Example: Update a field at the root level and a field in a nested level using the value selector.
+
+.Input
+[source,json,linenums]
+----
+{
+    "name": "Ken",
+    "lastName":"Shokida",
+    "age": 30,
+    "address": {
+        "street": "Amenabar",
+        "zipCode": "AB1234"
+    }
+}
+----
+
+.Source
+[source,weave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload update {
+    case age at .age -> age + 1
+    case s at .address.street -> "First Street"
```

Comment:
@machaval should we make use of `s` in the example or maybe just not feature it?